emcc.py

```diff
@@ -908,7 +911,7 @@ def optimizing(opts):
         file_suffix = get_file_suffix(arg)
         if file_suffix in SOURCE_ENDINGS + BITCODE_ENDINGS + DYNAMICLIB_ENDINGS + ASSEMBLY_ENDINGS + HEADER_ENDINGS or shared.Building.is_ar(arg): # we already removed -o <target>, so all these should be inputs
           newargs[i] = ''
-          if file_suffix.endswith(SOURCE_ENDINGS):
+          if file_suffix.endswith(SOURCE_ENDINGS) or (has_dash_c and file_suffix.endswith(BITCODE_ENDINGS)):
```

Comment:
Should we use `.is_bitcode()` here instead?  

Also, I'm not sure why I suggested dropping the WASM_OBJECT_FILES check here.  Does it make sense to compile bitcode to bitcode?  or should we just copy in that case?

Reply:
If you try to compile an already compiled object with clang it says " 'linker' input unused [-Wunused-command-line-argument]" so we should probably mirror that behaviour.